db.js

```diff
@@ -731,13 +775,73 @@ exports.init = function (sbot, config) {
           done((err) => {
             // prettier-ignore
             if (err) return unlock(cb, clarify(err, 'reindexEncrypted() failed to force-flush indexes'))
+            indexingActive.set(indexingActive.value - 1)
             unlock(cb)
           })
         })
       )
     })
   }
 
+  function notYetZero(obz, fn, ...args) {
+    if (obz.value > 0) {
+      onceWhen(obz, (x) => x === 0, fn.bind(null, ...args))
+      return true
+    } else {
+      return false
+    }
+  }
+
+  function compact(cb) {
+    if (notYetZero(jitdb.indexingActive, compact, cb)) return
+    if (notYetZero(jitdb.queriesActive, compact, cb)) return
+    if (notYetZero(indexingActive, compact, cb)) return
+
+    fs.closeSync(fs.openSync(resetLevelPath(dir), 'w'))
+    fs.closeSync(fs.openSync(resetPrivatePath(dir), 'w'))
+    fs.closeSync(fs.openSync(reindexJitPath(dir), 'w'))
```

Comment:
Yes, this is related to:

> What about browser compatibility for AAOL/compaction.js and DB2 post-compact files

I'll have to test this to be sure. Currently writing a spec for the new feed format, so will circle back to this later. For now I think this shouldn't be a show-stopper for this PR.

Reply:
Maybe could change these to atomic-file-rw

Reply:
Well there is also something in AAOL anyway so have to check both. So if you tested this and it works, fine to leave it.